server/src/main/java/org/apache/uniffle/server/buffer/ShuffleBufferManager.java

```diff
@@ -431,20 +436,28 @@ private Map<String, Set<Integer>> pickFlushedShuffle() {
     long expectedFlushSize = highWaterMark - lowWaterMark;
     long pickedFlushSize = 0L;
     int printIndex = 0;
+    int printIgnoreIndex = 0;
     int printMax = 10;
     for (Map.Entry<String, AtomicLong> entry : sizeList) {
       long size = entry.getValue().get();
-      pickedFlushSize += size;
       String appIdShuffleIdKey = entry.getKey();
-      addPickedShuffle(appIdShuffleIdKey, pickedShuffle);
-      // print detail picked info
-      if (printIndex < printMax) {
-        LOG.info("Pick application_shuffleId[{}] with {} bytes", appIdShuffleIdKey, size);
-        printIndex++;
-      }
-      if (pickedFlushSize > expectedFlushSize) {
-        LOG.info("Finish flush pick with {} bytes", pickedFlushSize);
-        break;
+      if (size > this.shuffleFlushThreshold) {
```

Comment:
If max size in `pickedShuffle` < `shuffleFlushThreshold`, the flushing mechanism is invalid? Right?

Reply:
I think there is no need to do such change. If too much small IO happened, we'd better to expand the cluster to distribute pressure.

Besides, the small IO could be observed by the metrics of `event_size_threshold_level4`

Reply:
> How to expand the cluster? more shuffle servers? There would be limited number of shuffle server and you cannot scale to many shuffle servers for one application.

Yes. More shuffle-servers could distribute small IO pressure from discrete usage of memory, especially if too much partitions are in one shuffle-server.

> This configurations is specially tailed for HDD only shuffle servers. I think it's necessary to reduce small I/Os to HDDs as much as possible.

If having this feature, possibly the small partition memory could be not flushed to persistent storage, and then all client will be back pressured.

> do you mean event_size_threshold_level1?

Yes

Reply:
Yes. Got your thought. But I consider an extreme case, if the all buffers size are 1M and memory reaches the high-watermark. After that, the later sending data requests are not the existing buffers' partition, whether the memory's data will be flushed?

Reply:
Emm.. If having this possibility, we should take consideration in our design instead of hoping not happening in users environment.